### DISCUSSION TITLE
Code documentation generation via Greenwood plugin

### DISCUSSION THREAD

**Discussion by hutchgrant** [ProjectEvergreen/greenwood#780] [Ideas]
What if we had a parser that could parse any(first with Greenwood) projects code and generate a Greenwood based documentation site ?

**Comment by thescientist13** [comment]
Are you specifically thinking of API documentation, or something else?

I think for someone wanting to create a Web Component library (single oackage or monorepo),  I could envision Greenwood definitely helping with creating a docs site for that repo, and also helping with auto generating API docs.  Probably not at its core per se, but maybe a plugin that takes advantage of the [custom elements analyzer](https://github.com/open-wc/custom-elements-manifest) project for the API docs specifically?

I think that is what [**ApolloElements**](https://github.com/apollo-elements/apollo-elements) uses, in combination with 11ty / Rocket static site generators.

I could also see integration opportunities if a project is using TypeScript (e.g. JS Docs)

----

This could eventually all be bundled into a template per work happening in #770 , whatever the outcome.

**Comment by thescientist13** [comment]
I think this would definitely be cool to explore, especially through our init package, and so maybe good to move this conversation over to a dedicated discussion, where I have included documentation site as one of the examples - https://github.com/ProjectEvergreen/greenwood/discussions/1335
